src/main/lighting/SpotLight.java

```diff
@@ -0,0 +1,91 @@
+package lighting;
+
+import math.compare.DoubleCompare;
+import primitives.Colour;
+import primitives.NormalizedVector;
+import primitives.Point;
+
+/**
+ * This class is represents an directional light source that attenuates with
+ * respect to distance.
+ *
+ *
+ * @author Abraham Murciano
+ * @author Eli Levin
+ */
+public class SpotLight extends PointLight {
+
+	private NormalizedVector direction;
+	private double divergence;
+	private double sharpness;
+
+	/**
+	 * Construct a {@link SpotLight} from a {@link Colour}, a direction, and 3
+	 * doubles that represent the attenuation constants.
+	 * 
+	 * @param colour    The colour of the light
+	 * @param position  The position in space of the light source
+	 * @param direction The direction of the spot light
+	 * @param q         The quadratic factor
+	 * @param l         The linear factor
+	 * @param c         The constant factor
+	 */
+	public SpotLight(Colour colour, Point position, double q, double l, double c, NormalizedVector direction) {
+		super(colour, position, q, l, c);
```

Comment:
Just call the long constructor instead of super so as not to repeat the setting of the extra 2 variables